packages/preferences-api/src/common/custom-keybinding-service.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { Disposable, DisposableCollection, CommandRegistry } from '@theia/core/lib/common';
+import { KeybindingEvent, KeybindingServer, JsonChange } from './keybindings-protocol';
+import * as Ajv from "ajv";
+
+
+export {
+    JsonChange
+};
+
+export const keybindingsSchema = {
+    "type": "array",
+    "items": {
+        "type": "object",
+        "properties": {
+            "keybinding": {
+                "type": "string"
+            },
+            "command": {
+                "type": "string"
+            },
+            "args": {
+                "type": "array"
+            }
+        },
+        "required": ["command", "keybinding"],
+        "optional": [
+            "args"
+        ],
+        "additionalProperties": false
+    }
+}
+
+@injectable()
+export class CustomKeybindingService implements Disposable {
+
+    protected readonly toDispose = new DisposableCollection();
+    protected readonly ajv = new Ajv();
+
+
+    protected resolveReady: () => void;
+    readonly ready = new Promise<void>(resolve => {
+        this.resolveReady = resolve;
+    });
+
+    constructor(
+        @inject(KeybindingServer) protected readonly server: KeybindingServer,
+        @inject(CommandRegistry) protected readonly commandRegistry: CommandRegistry
+    ) {
+        server.setClient({
+            onDidChangePreference: event => this.onDidChangePreference(event)
+        });
+        this.toDispose.push(server);
+    }
+
+    dispose(): void {
+        this.toDispose.dispose();
+    }
+
+    protected onDidChangePreference(event: KeybindingEvent): void {
+        for (const kbChange of event.changes) {
```

Comment:
this method should call  `KeybindingRegistry.setKeymap(keymap)` that in its turn should implement: https://github.com/theia-ide/theia/pull/456#issuecomment-327407558

Reply:
In that case shouldn't the RawKeyBinding interface be in the core? Or does keymap here reference something that's already declared in the core?